services/tier-service/src/main/java/com/amazon/aws/partners/saasfactory/saasboost/dal/ddb/DynamoTier.java

```diff
@@ -77,11 +78,11 @@ public Map<String, AttributeValue> updateAttributes() {
     }
 
     public Map<String, AttributeValue> primaryKey() {
-        return Map.of(PRIMARY_KEY.name(), attributes.get(PRIMARY_KEY.name()));
+        return Collections.singletonMap(PRIMARY_KEY.name(), attributes.get(PRIMARY_KEY.name()));
     }
 
     public static Map<String, AttributeValue> primaryKey(String id) {
-        return Map.of(PRIMARY_KEY.name(), AttributeValue.builder().s(id).build());
+        return Collections.singletonMap(PRIMARY_KEY.name(), AttributeValue.builder().s(id).build());
     }
 
     public static Tier fromAttributes(Map<String, AttributeValue> attributes) {
```

Comment:
Something to think about... In the other services, we wrap calls to the domain object setters defensively with a check to make sure the record coming back from DDB has a value for the field before we try to deserialize it. Granted, if the current data access layer created the serialized version, the current data access layer can deserialize it without error. But, when you update and have existing records in DDB you can get NPE and other exceptions.

For example:
```
if (item.containsKey("modified")) {
    try {
        LocalDateTime modified = LocalDateTime.parse(item.get("modified").s(), DateTimeFormatter.ISO_DATE_TIME);
        tenant.setModified(modified);
    } catch (DateTimeParseException e) {
        LOGGER.error("Failed to parse created date from database: " + item.get("modified").s());
        LOGGER.error(Utils.getFullStackTrace(e));
    }
}
```